homeassistant/components/pcf8574/config_flow.py

```diff
@@ -0,0 +1,84 @@
+"""Config flow for pcf8574 integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pcf8574 import PCF8574
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# TODO adjust the data schema to the data that you need
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("i2c_port_num", default=0): vol.Coerce(int),
+        vol.Required(
+            "i2c_address", default=0x20, description="0x20 hex is 32 dec"
+        ): vol.Coerce(
+            int
+        ),  # 0x20 = dec 32
+        vol.Optional("invert_logic", default=False): cv.boolean,
+        vol.Optional("switch_1_name"): cv.string,
```

Comment:
I am not familiar with I2C, is it always configured with 8 ports? Or did you just use this as a common example?

Reply:
The PCF8574 has 8 GPIO's

Reply:
yes the PCF8574 has exactly 8 GPIO's and this integration is specific for PCF8574. That's why i hardcoded it with 8 switches. 

in theory this integration can also be used as binary_sensor. Do you have any idea how I could write the config flow sequence for that ? 
I could create one boolean per GPIO to define `input` or `output`, but that would look messy ( 16 options on one page )  ? what do you think ?

Reply:
There is a nice blog post (https://aarongodfrey.dev/home%20automation/building_a_home_assistant_custom_component_part_3/) that shows how to loop config flow, so you could do a loop for each GPIO, but I'm not sure if this would be the best UX.
Maybe in the first step ask for the I2C address, port number, and the number of switches and binary sensors.
If the user will enter that he has at least 1 switch you show a form with inputs asking for switches names, if he entered that he has as least one binary sensor then a similar form would show but for binary sensors.
So you would always show the first form, then second, and third depending on values from the first step.
What do you think about such flow?

Reply:
thanks!
i've no split it up into 2 to 9 steps. 
step 1: i2c parameters
step 2 -8: configure each pin, and define input/output, name, invert_logic

each pin gets a unique id and therefore the name can be changed later. 
is it also possible to change the `invert_logic` parameter afterwards ?

Reply:
> is it also possible to change the invert_logic parameter afterwards ?

You can use OptionsFlow for that - https://developers.home-assistant.io/docs/config_entries_options_flow_handler/